crud/src/main/java/com/redhat/lightblue/rest/crud/cmd/CheckHealthCommand.java

```diff
@@ -0,0 +1,92 @@
+/*
+ Copyright 2013 Red Hat, Inc. and/or its affiliates.
+
+ This file is part of lightblue.
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.redhat.lightblue.rest.crud.cmd;
+
+import com.codahale.metrics.health.HealthCheck;
+import com.codahale.metrics.health.HealthCheck.Result;
+import com.codahale.metrics.health.HealthCheckRegistry;
+import com.redhat.lightblue.mediator.Mediator;
+import com.redhat.lightblue.rest.CallStatus;
+import com.redhat.lightblue.rest.crud.RestCrudConstants;
+import com.redhat.lightblue.rest.crud.health.CrudCheckRegistry;
+import com.redhat.lightblue.rest.crud.health.CrudCheckResponse;
+import com.redhat.lightblue.util.Error;
+import com.redhat.lightblue.util.metrics.RequestMetrics;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ *
+ * @author dhaynes
+ */
+public class CheckHealthCommand extends AbstractRestCommand {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CheckHealthCommand.class);
+
+    private RequestMetrics metrics;
+    HealthCheckRegistry healthCheckRegistry;
+
+    public CheckHealthCommand(HealthCheckRegistry healthCheckRegistry, RequestMetrics metrics) {
+        this(null, healthCheckRegistry, metrics);
+    }
+
+    public CheckHealthCommand(Mediator mediator, HealthCheckRegistry healthCheckRegistry, RequestMetrics metrics) {
+        super(mediator);
+        this.healthCheckRegistry = healthCheckRegistry;
+        this.metrics = metrics;
+    }
+
+    @Override
+    public CallStatus run() {
+
+        RequestMetrics.Context context = metrics.startHealthRequest();
```

Comment:
I agree it's worth the small cost to capture these metrics. It is worth remembering I think though that these will run async, so this won't actually time how long it takes to run the health checks themselves, even though that may still be useful information.

This makes me think two things:

1. CRUDHealth is probably a little _too_ basic. The service health we are doing internally has arbitrary map (key/value) parameters possibly, matching dropwizard's capabilities (in other words, dropwizard's health checks can add these parameters easily from the CRUDHealth results for example). Then we could neatly add like, what mongo configuration was used and stuff, instead of cramming it all into one string field.
2. In that same vein, CRUDHealth could add a duration field if we wanted (thus getting that more useful timing information). It wouldn't be in Graphite, but it would at least be a part of the diagnostics response for someone to take a peak at.

That said, I don't necessarily think either of those things are essential, though they may be relatively easy to add.

If we wanted to get fancy we could probably add real JMX metric tracking to the health checks, too, maybe by passing the RequestMetrics instance to the getHealthRegistry methods. Anyway, it's a bit extra, so use your judgement of course :-).